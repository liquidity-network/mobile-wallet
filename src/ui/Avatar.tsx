import React, { PureComponent } from 'react'
import { View, Image, Text } from 'react-native'

import { ACTIVE_COLOR, globalStyles, WHITE } from 'globalStyles'

interface Props {
  url?: string
  name?: string
  small?: boolean
}

export default class Avatar extends PureComponent<Props> {
  render() {
    const containerStyles = this.props.small ? styles.containerSmall : styles.container
    return this.props.url != null ? (
      <Image source={{ uri: this.props.url }} style={containerStyles} />
    ) : (
      <View style={containerStyles}>
        {this.props.name != null && (
          <Text style={this.props.small ? styles.initialsSmall : styles.initials}>
            {getNameInitials(this.props.name)}
          </Text>
        )}
      </View>
    )
  }
}

const styles: any = {
  container: {
    width: 48,
    height: 48,
    borderRadius: 24,
    backgroundColor: ACTIVE_COLOR,
    alignItems: 'center',
    justifyContent: 'center',
  },
  initials: { ...globalStyles.P_BIG, color: WHITE },
}
styles.containerSmall = { ...styles.container, width: 32, height: 32, borderRadius: 16 }
styles.initialsSmall = { ...styles.initials, fontSize: 14 }

function getNameInitials(name) {
  const words = name.split(' ')
  return `${words[0] ? words[0][0].toUpperCase() : ''}${words[1] ? words[1][0].toUpperCase() : ''}`
}
