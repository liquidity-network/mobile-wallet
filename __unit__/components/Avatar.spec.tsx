import React from 'react'
import { create } from 'react-test-renderer'

import Avatar from 'ui/Avatar'

describe('<Avatar> component', () => {
  it('renders w/url correctly', () => {
    const tree = create(<Avatar url={'https://google.com'} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders w/name correctly', () => {
    const tree = create(<Avatar name="Michael Jackson" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
