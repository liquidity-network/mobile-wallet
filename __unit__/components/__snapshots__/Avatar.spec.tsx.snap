// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar> component renders w/name correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5436cd",
      "borderRadius": 24,
      "height": 48,
      "justifyContent": "center",
      "width": 48,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "AktivGroteskCorp-Regular",
        "fontSize": 18,
      }
    }
  >
    MJ
  </Text>
</View>
`;

exports[`<Avatar> component renders w/url correctly 1`] = `
<Image
  source={
    Object {
      "uri": "https://google.com",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#5436cd",
      "borderRadius": 24,
      "height": 48,
      "justifyContent": "center",
      "width": 48,
    }
  }
/>
`;
